metadata-ingestion/src/datahub/ingestion/source/iceberg/iceberg.py

```diff
@@ -608,12 +613,15 @@ def get_report(self) -> SourceReport:
         return self.report
 
     def _create_iceberg_namespace_aspects(
-        self, namespace: Identifier
+        self, namespace: Identifier, properties: Properties
     ) -> Iterable[_Aspect]:
         namespace_repr = ".".join(namespace)
         yield Status(removed=False)
         yield ContainerProperties(
-            name=namespace_repr, qualifiedName=namespace_repr, env=self.config.env
+            name=namespace_repr,
+            qualifiedName=namespace_repr,
+            env=self.config.env,
+            customProperties=properties,
```

Comment:
are properties always a Dict[str, str]? if not, we need to coerce it to be that

something like Dict[str, Any] might also cause issues

Reply:
That's interesting. `properties` are of type `Properties` which is defined in `pyiceberg`. In `<=0.4` it was defined as `Dict[str, str]` - so I just used it as it is, but now I see in [newer versions](https://github.com/apache/iceberg-python/blame/main/pyiceberg/typedef.py#L89) it is `Dict[str, Any]`. To be forward compatible, I will adjust it to make all values (and maybe keys) as strings.